onboarding_auth/onboardingAuthPresentation/src/main/java/com/alexafit/onboardingAuthPresentation/register/RegisterViewModel.kt

```diff
@@ -0,0 +1,124 @@
+package com.alexafit.onboardingAuthPresentation.register
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.alexafit.core.util.UiEvent
+import com.alexafit.core.util.UiText
+import com.alexafit.onboardingAuthPresentation.R
+import com.alexafit.onboardingAuthPresentation.event.user.RegisterUserEvent
+import com.alexafit.onboardingauthdomain.useCase.OnboardingAuthUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class RegisterViewModel @Inject constructor(
+    private val onboardingAuthUseCase: OnboardingAuthUseCase
+) : ViewModel() {
+
+    var registerState by mutableStateOf(RegisterState())
+        private set
+
+    private val _uiEvent = Channel<UiEvent>()
+    val uiEvent = _uiEvent.receiveAsFlow()
+
+    fun onEvent(event: RegisterUserEvent) {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
There shouldn't be any need for replacing the dispatcher in a ViewModel. It should always be the job of the suspending function to assure it's safe to be called from the main thread (by using `withContext(dispatcher)` there).
So only if you implement a suspending function which calls non-suspending blocking functions, you need to switch the dispatcher like that. Otherwise all the suspending functions from Retrofit, Room, etc. will switch under the hood.